app/Controllers/Auth/AuthController.php

```diff
@@ -26,41 +26,104 @@ protected function checkRecaptcha(ValidationHelper $validator, Request $request)
 
 
     /**
-     * @return bool|false|resource
+     * Connects to LDAP server and logs in with service account (if configured)
+     * @return resource|false
      */
     public function ldapConnect()
     {
         if (!extension_loaded('ldap')) {
             $this->logger->error('The LDAP extension is not loaded.');
             return false;
         }
-
-        $server = ldap_connect($this->config['ldap']['host'], $this->config['ldap']['port']);
-
+        // Building LDAP URI
+        $ldapSchema=(@is_string($this->config['ldap']['schema'])) ?
+            strtolower($this->config['ldap']['schema']) : 'ldap';
+        $ldapURI="$ldapSchema://".$this->config['ldap']['host'].':'.$this->config['ldap']['port'];
+        
+        // Connecting to LDAP server
+        $this->logger->debug("Connecting to $ldapURI");
+        $server = ldap_connect($ldapURI);
         if ($server) {
             ldap_set_option($server, LDAP_OPT_PROTOCOL_VERSION, 3);
             ldap_set_option($server, LDAP_OPT_REFERRALS, 0);
             ldap_set_option($server, LDAP_OPT_NETWORK_TIMEOUT, 10);
+        } else {
+            $this->logger->error(ldap_error($server));
+            return false;
+        }
+        
+        // Upgrade to StartTLS
+        $useStartTLS = @is_bool($this->config['ldap']['useStartTLS']) ? $this->config['ldap']['useStartTLS'] : false;
```

Comment:
Is really needed the check on the `@is_bool` ? Should be better to check the existance of the key using `isset()`

Reply:
I  use the is_{type} function also to if the input is the correct type along with testing if the array index is set. It is important to validate because not all system administrators are familiar with PHP types and may mix types in the config.php file. I just added the '@' to the is_{type} function to quiet the PHP warning this is expected when the array index isn't set.

Reply:
> I use the is_{type} function also to if the input is the correct type along with testing if the array index is set. It is important to validate because not all system administrators are familiar with PHP types and may mix types in the config.php file. I just added the '@' to the is_{type} function to quiet the PHP warning this is expected when the array index isn't set.

Sure, but I think is better to have the application failing loudly that silently taking another input if the configuration is incorrect.
Using the `is_{type}` functions with the `@` gives you any hint in case something goes wrong.